2-Draft-Accepted/RFCNNNN-Support-Experimental-Features.md

```diff
@@ -457,6 +470,27 @@ as well as a problem that we have to solve before using the experimental feature
 
 > Not sure how to handle it yet ...
 
+### Experimental feature incompatibility mitigations
+There will be cases where experimental features are incompatible either intentionally or unintentionally.
+
+* An example of an intentional incompatibility occurs when two alternative, but incompatible solutions, are being tested.
+* An example of an unintended incompatibility can occur when two experimental features prove to be incompatible through usage.
+
+A minimal implementation will support defining a list of experimental feature pairs in the `powershell.config.json` file.  At startup, attempts to use incompatible features would produce warnings and consider both as disabled.
+
+An opportunistic implementation will support an exclusion value in the experimental feature declaration for cases where an author is intentionally providing alternative/incompatible implementations.
+
+### Static Code Analysis
+Static code analysis support should be provided to address experimental feature authoring. Two areas are particularily important:
+
+1: Validating an experimental feature is marked correctly.
+A use case for this occurs when an experimental feature is referenced in a module but not, or incorrectly defined in the manifest.  From a script perspective, a PSScriptAnalyzer rule would be a reasonable solution. For a compiled code perspective, a reflection-based tool may be needed.
+
+2: Validating references to an experimental feature have been removed.
+At some point, the experiment is completed and either removed from or incorporated into the code base. At this point, a tool to scan script or compiled code would be needed to ensure any references to the experiment have been removed.
+
+> NOTE: Will consumers require builds with no experimental code? If so, how to do we ensure this?
```

Comment:
I think it may be sufficient to have policy settings so users cannot enable experimental code if there are concerns for enterprises.

Reply:
While that works for enterprise, it doesn't address cloud.  TBH: I'm more interested in ensuring experiments are declared correctly when added and dead code is removed when the experiment is complete.

Reply:
Policy settings can be set in `powershell.config.json` and take effect without GPO.
On windows, the same policy settings in registry (GPO) take preference over the settings in `powershell.config.json`.